src/main/java/com/artipie/asto/fs/VertxFileStorage.java

```diff
@@ -0,0 +1,243 @@
+            }).to(SingleInterop.get()).toCompletableFuture();
+    }
+
+    @Override
+    public CompletableFuture<Void> save(final Key key, final Content content) {
+        return Single.fromCallable(
+            () -> {
+                final Path tmp = Paths.get(
+                    this.dir.toString(),
+                    String.format("%s.%s.tmp", key.string(), UUID.randomUUID())
+                );
+                tmp.getParent().toFile().mkdirs();
+                return tmp;
+            })
+            .subscribeOn(RxHelper.blockingScheduler(this.vertx.getDelegate()))
+            .flatMapCompletable(
+                tmp -> new VertxRxFile(
+                    tmp,
+                    this.vertx
+                ).save(Flowable.fromPublisher(content))
+                    .andThen(
+                        this.vertx.fileSystem()
+                            .rxMove(
+                                tmp.toString(),
+                                this.path(key).toString(),
+                                new CopyOptions().setReplaceExisting(true)
+                            )
+                    )
+                    .onErrorResumeNext(
+                        throwable -> new VertxRxFile(tmp, this.vertx)
+                            .delete()
+                            .andThen(Completable.error(throwable))
+                    )
+            )
+            .to(CompletableInterop.await())
+            .<Void>thenApply(o -> null)
+            .toCompletableFuture();
+    }
+
+    @Override
+    public CompletableFuture<Void> move(final Key source, final Key destination) {
+        return Single.fromCallable(
+            () -> {
+                final Path dest = this.path(destination);
+                dest.getParent().toFile().mkdirs();
+                return dest;
+            })
+            .flatMapCompletable(
+                dest -> new VertxRxFile(this.path(source), this.vertx).move(dest)
+            )
+            .to(CompletableInterop.await())
+            .<Void>thenApply(file -> null)
+            .toCompletableFuture();
+    }
+
+    @Override
+    public CompletableFuture<Void> delete(final Key key) {
+        return new VertxRxFile(this.path(key), this.vertx)
+            .delete()
+            .to(CompletableInterop.await())
+            .toCompletableFuture()
+            .thenCompose(ignored -> CompletableFuture.allOf());
+    }
+
+    @Override
+    public CompletableFuture<Long> size(final Key key) {
+        return Single.fromCallable(
+            () -> {
+                try {
+                    return Files.size(this.path(key));
+                } catch (final NoSuchFileException nofile) {
+                    throw new ValueNotFoundException(key, nofile);
+                } catch (final IOException iex) {
+                    throw new UncheckedIOException(iex);
+                }
+            }
+        ).subscribeOn(RxHelper.blockingScheduler(this.vertx.getDelegate()))
```

Comment:
@Sammers21 many others methods in this class run blocking operations using `Files`. May we use same blocking scheduler for them as well?